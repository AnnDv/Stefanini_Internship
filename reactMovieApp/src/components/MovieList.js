import React from 'react';
import PropTypes from 'prop-types';
import MovieImage from './movieComponents/MovieImage';
import MovieTitle from './movieComponents/MovieTitle';
import './MovieList.css';
import ShowInfoButton from '../features/infoButton/ShowInfoButton';

function MovieList({ movies }) {
  return (
    <div className='grid'>
      {movies?.map((movie) => (
        <div key={movie.id} className="card_description">
          <MovieImage image={movie.poster_path}/>
          <MovieTitle title={movie.title}/>
          <ShowInfoButton overview={movie.overview}/>
        </div>
      ))}
    </div>
  );
}

MovieList.propTypes = {
  movies: PropTypes.array,
};

export default MovieList;
