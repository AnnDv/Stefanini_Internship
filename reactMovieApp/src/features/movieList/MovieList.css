.grid{
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;
}

.card_description {
    padding: 20px;
}
