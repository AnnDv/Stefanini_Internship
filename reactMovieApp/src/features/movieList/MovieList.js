import React from 'react';
import PropTypes from 'prop-types';
import './MovieList.css';
import MovieCard from '../movieCard/MovieCard';

function MovieList({ movies }) {
  return (
    <div data-testid="list" className='grid'>
      {movies?.results?.map((movie) => (
        <div key={movie.id} data-testid={`m-${movie.id}`} className="card_description">
          <MovieCard movie={movie}/>
        </div>
      ))}
    </div>
  );
}

MovieList.propTypes = {
  movies: PropTypes.object,
};

export default MovieList;
