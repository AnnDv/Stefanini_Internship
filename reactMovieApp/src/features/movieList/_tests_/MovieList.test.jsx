import React from 'react';
import '@testing-library/jest-dom';
import { render, waitFor } from '@testing-library/react';
import movies from '../_fixtures_/movies';
import MovieList from '../MovieList.jsx';

describe('MovieList', () => {
  const { getByTestId, baseElement } = render(<MovieList movies={movies}/>);
  it('should match the snapshot', async () => {
    await waitFor(async () => {
      const movieColl = getByTestId('list').children[0];
      console.log('teststst  ', movieColl);
      const movieCollLength = movieColl;
      expect(movieCollLength).toBeGreaterThan(0);
    });
    expect(baseElement).toMatchSnapshot();
  });
});
