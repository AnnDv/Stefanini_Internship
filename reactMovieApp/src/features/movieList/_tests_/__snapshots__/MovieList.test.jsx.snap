// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieList should match the snapshot 1`] = `
<body>
  <div>
    <div
      class="grid"
      data-testid="list"
    />
  </div>
</body>
`;
